src/OpenTelemetry.ResourceDetectors.Azure/AzureVmMetaDataRequestor.cs

```diff
@@ -0,0 +1,40 @@
+// <copyright file="AzureVmMetaDataRequestor.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Net.Http;
+using System.Text.Json;
+
+namespace OpenTelemetry.ResourceDetectors.Azure;
+
+internal sealed class AzureVmMetaDataRequestor : IAzureVmMetaDataRequestor
```

Comment:
This can be a static class. Is the interface added just for the unit test? If yes, then we can do the unit test even without the interface.